app/customer.py

```diff
@@ -0,0 +1,58 @@
+import dataclasses
+import math
+from typing import Any
+
+
+@dataclasses.dataclass
+class Customer:
+    name: str
+    location: list
+    money: int
+    car: dict
+    product_cart: dict
+
+    @staticmethod
+    def fuel_to_shop(
+            fuel_cons: float, distance_to_shop: float, fuel_price: float
+    ) -> float:
+        return ((fuel_cons * distance_to_shop) / 100 * fuel_price) * 2
+
+    def dist_to_shop(self, shop_location: list) -> float:
+        return (
+            math.sqrt((shop_location[0] - self.location[0]) ** 2
+                      + (shop_location[1] - self.location[1]) ** 2)
+        )
+
+    def product_total(self, products: dict) -> int:
+        prod_total = 0
+        for key_1, value_1 in products.items():
+            for key_2, value_2 in self.product_cart.items():
+                if key_1 == key_2:
+                    prod_total += value_1 * value_2
+        return prod_total
```

Comment:
Now is more better. Just change a little by using sum and [] 
```suggestion
        return sum(
            count * products[name]
            for name, count in self.product_cart.items()
        )
```